### DISCUSSION TITLE
What is the difference from devspace?

### DISCUSSION THREAD

**Discussion by liyimeng** [acorn-io/runtime#622] [General]
Interesting project! How it is different from devSpace, builtin registry? better workflow?

**Comment by cjellick** [comment]
This is a great question. We are going to do one or more blog diving deeper into the differences between this and other tools.

They do both aim to ease the development workflow related to Kuberentes, but diverge quickly beyond that. 

I'm not a devSpace expert, but here's my take:

At the end of the day, devSpace still wants you creating/deploying/managing the native kubernetes resources (deployments, ingress, services, etc). In their own words devSpace is "kubectl on steroids". This gives you access to the entire k8s API space, which offers maximum flexibility, but also maximum complexity.

With Acorn, users don't need any Kubernetes knowledge. It is an abstraction layer on top of Kubernetes with its own application-level constructs. We are curating our users' experience to make it less complex.

**Comment by liyimeng** [comment]
@cjellick Thanks for explanation. In this case, would some ide work should be considered, e.g. have a vscode  acorn plugin ?

**Comment by cjellick** [comment]
Saw you opened an issue as well. Will give the same question here: you mean for simple syntax highlighting of the Acornfile or did you have something else in mind?

**Comment by liyimeng** [comment]
I don't have much in mind.  That would be something to start with. Acornfile is in json, readability will become an issue when it grows big.  However, I think most IDE already support folding  json file already.  Define workflow, auto-completion could be something to consider? 

I would close the issue since I thought it is less active here. :)

**Reply by vincent99** [reply]
For now if you want folding and such the closest option is installing tooling for cuelang and telling your IDE that it's a cue file.

Yes I would like there to be vscode/vim/etc plugins that know all the fields that are valid in an Acornfile and can suggest them, etc.  There is not currently anybody working on that though, partially because it's still early and the language is still evolving.

**Comment by cjellick** [comment]
+1 to what Vince said.

Leave the issue open please and I'll just cross reference to here. We keep our eyes on both forums as best we can, but we use issues to drive/track work.

**Comment by cloudnautique** [comment]
Plugin has been published to the market place, https://marketplace.visualstudio.com/items?itemName=Acorn-Labs.acorn

Please open issues and feature requests in github.com/acorn-io/vscode-acorn
